Shared state shows test change in the wrong action

rcarver opened the pull request:
When modifying a `@Shared` value in a delegate action, the `TestStore` shows the state change in the original action, not the delegate. 

This PR is a failing test only. In reality this would be a parent feature responding to the delegate, but the test works within a single reducer.

mbrandonw commented:
Hi @rcarver, this is a known, but also expected, issue with shared state and the test store. The fix is to have the test store not eagerly process actions from effects, but unfortunately that would be a breaking change. And so we will have to wait until 2.0 to fix that.

For now you will just need to assert on shared state in the `received`, or you can also do it after the `send`:

```swift
await store.send(.incrementSharedInDelegate)
store.state.$stats.assert {
  $0.count = 1
}
await store.receive(\.delegate.didIncrement) {
  $0.count = 1
}
```

We would be happy to accept this PR if you want to mark this test as an expected failure. You can search the repo for `XCTTODO` to see how we have done this in a few spots in the code base. That will help us keep track of this when we finally fix the problem.

rcarver commented:
Thanks @mbrandonw. I added `XCTTODO` to the existing test, and also added a new test showing the behavior today. 

FYI the version you proposed doesn't pass for me

```swift
await store.send(.incrementSharedInDelegate)
store.state.$stats.assert {
  $0.count = 1
}
await store.receive(\.delegate.didIncrement) {
  $0.count = 1
}
```

But this form does

```swift
    await store.send(.incrementSharedInDelegate) {
      $0.stats.count = 1
    }
    await store.receive(\.delegate.didIncrement) {
      $0.count = 1
    }
```

mbrandonw commented:
> FYI the version you proposed doesn't pass for me

Oh, huh, do you remember what the error was?

rcarver commented:
@mbrandonw oh sorry I meant to add that, here you go:

<img width="781" alt="image" src="https://github.com/user-attachments/assets/9840fbc4-14ce-426e-888a-98f0b29afe89">
